analyzer/src/main/kotlin/managers/Cargo.kt

```diff
@@ -90,10 +90,9 @@ class Cargo(
         VcsHost.toVcsInfo(extractRepositoryUrl(node))
 
     private fun extractDeclaredLicenses(node: JsonNode) =
-        node["license"].textValueOrEmpty().split("/")
-            .map { it.trim() }
-            .filter { it.isNotEmpty() }
-            .toSortedSet()
+        sortedSetOf<String>().apply {
+            add(node["license"].textValueOrEmpty().replace("/", " OR ").trim())
+        }
```

Comment:
This could theoretically be written as `sortedSetOf(node["license"].textValueOrEmpty().replace("/", " OR ").trim())`, but I actually liked the trimming of the individual elements in the original code.

Anyway, I believe we have a more fundamental API problem here: The return value of this function is used to populate the `declaredLicenses` field, and that's defined as `val declaredLicenses: SortedSet<String>`. Strictly speaking we somewhat the same problem with the individual entries in the set: It's unclear whether set members have an `AND` or `OR` relation.

IIRC we deliberately did not *want* to chose *any* relation here, but `declaredLicenses` should literally take whatever string is used in the package manager's meta data. And as some package managers have multiple license fields, `declaredLicenses` is a collection / set instead of a single string.

Long story short, I'm not sure if even the original split by `/` was correct, as that's already not literally taking the meta data. I believe this needs some discussion.

Reply:
Well, what is there to discuss? I just read the whole story and in the Rust ecosystem the `/` was used as a `OR` in a SPDX license expression, which is what my commit tries to address.

If you believe the toolkit should get the raw value, and it can handle it, I can change it so we don't process anything here. In the meantime, I'd argue that it's better to cover more cases (correctly) over anything else.

Reply:
> It's unclear whether set entries have an `AND` or `OR` relation among each other.

Actually, looking again at how we generate `declaredLicensesProcessed`, I realize [we *do* relate set entries always with `AND`](https://github.com/oss-review-toolkit/ort/blob/376555874ab7e46a95e894a2702e15b94326ee85/model/src/main/kotlin/Package.kt#L60-L64).

Reply:
To conclude my thinking-out-aloud: I'd be fine with the intent of this change, but I'd implement it as

    private fun extractDeclaredLicenses(node: JsonNode) =
        sortedSetOf(
            node["license"].textValueOrEmpty().split('/')
                .map { it.trim() }
                .filter { it.isNotEmpty() }
                .joinToString(" OR ")
        )

Let's see what others say.

Reply:
Basically I agree with interpreting the `/` as `or`.
I wonder how the `and` operator looks like in Rust - any idea?

Reply:
@fviernau it doesn't exist, which is why the recommendation was to start using SPDX expressions instead.

Reply:
@sschuberth I have applied your suggestion.